File: packages/sdks/typescript/tester-local/__tests__/test_start_doc_query.ts
import * as dotenv from "dotenv";
import * as path from "path";
import { MetriportClient } from "../src";
import { StartDocumentQueryRequest } from "../src/api/resources/medical/resources/document/client/requests";


dotenv.config({ path: path.resolve(__dirname, '.env') });

const apiUrl = process.env.BASE_URL;
const apiToken = process.env.API_KEY;
const patientId = process.env.PATIENT_ID;
const facilityId = process.env.FACILITY_ID;

if (!apiUrl || !apiToken || !patientId || !facilityId) {
  throw new Error("Required environment variables are not set");
}

const client = new MetriportClient({
  environment: () => apiUrl,
  apiKey: () => apiToken,
});

describe("Document Query tests", () => {
  test("start document query", async () => {
    console.log("Calling start_query...");
    const requestData: StartDocumentQueryRequest = {
        facilityId: facilityId,
        patientId: patientId,
    }
    const response = await client.medical.document.startQuery(requestData);
    console.log(`response: ${JSON.stringify(response, null, 2)}`);
  });
});

File: packages/sdks/python/tester-local/tests/test_start_doc_query.py
import os

from generated.client import Metriport
from generated import commons
from generated.resources import medical

import os
from dotenv import load_dotenv

load_dotenv()

api_key = os.environ.get("API_KEY")
patient_id = os.environ.get("PATIENT_ID")
facility_id = os.environ.get("FACILITY_ID")
base_url = os.environ.get("BASE_URL")


def test_start_doc_query() -> None:
    client = Metriport(api_key=api_key, base_url=base_url)
    response = client.medical.document.start_query(patient_id=patient_id, facility_id=facility_id)
    print(f"Response: {response}")
